cli/src/main.rs

```diff
@@ -0,0 +1,88 @@
+// Copyright 2023 CeresDB Project Authors. Licensed under Apache-2.0.
+
+use std::path::Path;
+
+use async_trait::async_trait;
+use clap::Parser;
+use futures::executor::block_on;
+use sqlness::{
+    database_impl::mysql::MysqlDatabase, ConfigBuilder, DatabaseConfig, DatabaseConfigBuilder,
+    EnvController, Runner,
+};
+
+#[derive(Parser, Debug)]
+#[clap(author, version, about, long_about = None)]
+/// A cli to run sqlness tests.
+struct Args {
+    /// Directory of test cases
+    #[clap(short, long, required(true))]
+    case_dir: String,
+
+    /// IP of database to test against
+    #[clap(short, long, required(true))]
+    ip: String,
+
+    /// Port of database to test against
+    #[clap(short, long, required(true))]
+    port: u16,
+
+    /// User of database to test against
+    #[clap(short, long, required(false))]
+    user: Option<String>,
+
+    /// Password of database to test against
+    #[clap(short('P'), long, required(false))]
+    password: Option<String>,
+
+    /// DB name of database to test against
+    #[clap(short, long)]
+    db: Option<String>,
+
+    /// Which DBMS to test against, only mysql is supported now
+    #[clap(short, long, default_value("mysql"))]
+    r#type: String,
```

Comment:
We can change this to an enum:

https://docs.rs/clap/latest/clap/trait.ValueEnum.html

https://stackoverflow.com/questions/73878442/rust-clap-use-enum-for-separate-flags

Reply:
Sound goods.